arcgis-ios-sdk-samples/Layers/Query with CQL filters/QueryWithCQLFiltersSettingsViewController.swift

```diff
@@ -0,0 +1,171 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+protocol QueryWithCQLFiltersSettingsViewControllerDelegate: AnyObject {
+    func settingsViewController(_ controller: QueryWithCQLFiltersSettingsViewController, queryParameters: AGSQueryParameters)
+}
+
+class QueryWithCQLFiltersSettingsViewController: UITableViewController {
+    // MARK: Storyboard views
+    
+    /// The cell for where clause options.
+    @IBOutlet var whereClauseCell: UITableViewCell!
+    /// The cell for max features input.
+    @IBOutlet var maxFeaturesCell: UITableViewCell!
+    /// The start date picker.
+    @IBOutlet var startDatePicker: UIDatePicker!
+    /// The end date picker.
+    @IBOutlet var endDatePicker: UIDatePicker!
+    /// The switch to control if date pickers are enabled.
+    @IBOutlet var dateFilterSwitch: UISwitch!
+    
+    // MARK: Properties
+    
+    /// A delegate to notify other view controllers.
+    weak var delegate: QueryWithCQLFiltersSettingsViewControllerDelegate?
+    
+    let numberFormatter = NumberFormatter()
+    let sampleWhereClauses = [
+        // Sample Query 1, cql-text, cql-json.
+        "F_CODE = 'AP010'",
+        "{\"eq\":{\"property\":\"F_CODE\",\"value\":\"AP010\"}}",
+        // Sample Query 2.
+        "F_CODE LIKE 'AQ%'",
+        "{\"like\":{\"property\":\"F_CODE\",\"value\":\"AQ%\"}}",
+        // Sample Query 3: use cql-json to combine "before" and "after" temporal
+        // operators with the logcial "and" operator
+        "{\"and\":[{\"after\":{\"property\":\"ZI001_SDV\", \"value\":\"2011-12-31\"}},{\"before\":{\"property\":\"ZI001_SDV\",\"value\":\"2013-01-01\"}}]}"
```

Comment:
Note: The json query strings here are known to return 400 Bad Request. Zack and Hisham are looking into it. I'll keep this thread posted when I have updates.

Reply:
Actually, I'm thinking if we should remove the jsons here for now, and make another PR to add them when they get fixed. So that this PR won't be blocked. 🤔  What do you think?

Reply:
Removed for now, and will add back once `completion(Zack's new CQL jsons)` 🤣